test/compilable/cppmangle3.d

```diff
@@ -34,5 +34,22 @@ struct Foo
 alias Alias(alias a) = a;
 alias Alias(T) = T;
 
-static assert(is(Alias!(__traits(parent, bar)) == Foo));
```

Comment:
How did this originally pass? 
also `static assert(is(Alias!(__traits(parent, Foo.bar)) == Foo));` currently fails
as does `Foo.bar` (with `Error: no property bar for type Foo`). You should update the bugzilla issue or create a new one for this.

Reply:
https://github.com/dlang/dmd/pull/8721

Reply:
I mean How can you even refer to `bar` without going through `Foo`?

Reply:
The symbol resolution code using `toParent3` was a bit to eager I guess.
I tried reverting that PR code fully, but it fails on this test case.
I'll try to deprecate non-mangle `extern(C++, scoped)` soon. There's a tons of bugs in the implementation (e.g. it shows up in `__traits(allMembers)` which prevents me from automatically checking if my bindings are correct...).

Reply:
Awesome, looking forward to it.